File: clients/android-new/demo/src/main/java/cy/ac/ucy/cs/anyplace/android/SettingsActivity.java
package cy.ac.ucy.cs.anyplace.android;

import android.content.Context;
import android.content.Intent;
import android.net.Uri;
import android.os.Bundle;
import android.util.Log;
import android.view.View;

import androidx.appcompat.app.ActionBar;
import androidx.appcompat.app.AppCompatActivity;
import androidx.preference.Preference;
import androidx.preference.PreferenceFragmentCompat;

public class SettingsActivity extends AppCompatActivity {
  private final String TAG = SettingsActivity.class.getSimpleName();
  @Override
  protected void onCreate(Bundle savedInstanceState) {
    super.onCreate(savedInstanceState);
    setContentView(R.layout.settings_activity);
    getSupportFragmentManager()
            .beginTransaction()
            .replace(R.id.settings, new SettingsFragment())
            .commit();
    ActionBar actionBar = getSupportActionBar();
    if (actionBar != null) {
      actionBar.setDisplayHomeAsUpEnabled(true);
    }

  }


  public static class SettingsFragment extends PreferenceFragmentCompat {
    private static final String TAG = SettingsFragment.class.getSimpleName() ;

    @Override
    public void onCreatePreferences(Bundle savedInstanceState, String rootKey) {
        Log.d(TAG, rootKey +"");
        //setPreferencesFromResource(R.xml.root_preferences, rootKey);
        setPreferencesFromResource(R.xml.preferences_logger, rootKey);





      }




  }
}

File: clients/android-new/logger/src/main/java/cy/ac/ucy/cs/anyplace/logger/SettingsActivity.kt
package cy.ac.ucy.cs.anyplace.logger

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import cy.ac.ucy.cs.anyplace.lib.android.LOG.Companion.D

class SettingsActivity : AppCompatActivity() {
  private val TAG = SettingsActivity::class.java.simpleName
  override fun onCreate(savedInstanceState: Bundle?) {
    super.onCreate(savedInstanceState)
    setContentView(R.layout.settings_activity)
    supportFragmentManager
            .beginTransaction()
            .replace(R.id.settings, SettingsFragment())
            .commit()
    val actionBar = supportActionBar
    actionBar?.setDisplayHomeAsUpEnabled(true)
  }

  class SettingsFragment : PreferenceFragmentCompat() {
    fun onCreatePreferences(savedInstanceState: Bundle?, rootKey: String) {
      D(TAG, "root key: $rootKey")
      //setPreferencesFromResource(R.xml.root_preferences, rootKey);
      setPreferencesFromResource(R.xml.preferences_anyplace, rootKey)
    }

    companion object {
      private val TAG = SettingsFragment::class.java.simpleName
    }
  }
}